### PR TITLE
SmallDeformation gravity.

### PR DESCRIPTION
**Pull request by endJunction**
Closes https://github.com/ufz/ogs/issues/1945

This adds gravity to the small deformation process.

TODO:

- [x] Benchmark with non-zero gravity.
- [x] Optional benchmark with non-uniform solid density.

### CODE REVIEW

File: ProcessLib/SmallDeformation/CreateSmallDeformationProcess.cpp [hunk 1]
```diff
@@ -97,8 +97,32 @@ createSmallDeformationProcess(
...
+        "solid_density", parameters, 1);
+    DBUG("Use \'%s\' as solid density parameter.", solid_density.name.c_str());
+
+    // Specific body force
+    Eigen::Matrix<double, DisplacementDim, 1> specific_body_force;
+    {
+        std::vector<double> const b =
+            //! \ogs_file_param{prj__processes__process__SMALL_DEFORMATION__specific_body_force}
+            config.getConfigParameter<std::vector<double>>(
+                "specific_body_force");
```
Comment (chleh) [inline ProcessLib/SmallDeformation/CreateSmallDeformationProcess.cpp]: Why don't we just make everything a Parameter?

File: ProcessLib/SmallDeformation/SmallDeformationFEM.h [hunk 2]
```diff
@@ -185,6 +185,18 @@ class SmallDeformationLocalAssembler
             auto const& N = _ip_data[ip].N;
             auto const& dNdx = _ip_data[ip].dNdx;
 
+            typename ShapeMatricesType::template MatrixType<DisplacementDim,
+                                                            displacement_size>
+                N_u_op = ShapeMatricesType::template MatrixType<
```
Comment (chleh) [inline ProcessLib/SmallDeformation/SmallDeformationFEM.h]: Some people call that H matrix, if I'm right (learned that in Dublin :smile:)

File: ProcessLib/SmallDeformation/SmallDeformationProcessData.h [hunk 3]
```diff
@@ -47,6 +56,12 @@ struct SmallDeformationProcessData
 
     std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
         material;
+    /// Solid's density. A scalar quantity, Parameter<double>.
+    Parameter<double> const& solid_density;
+    /// Specific body forces applied to solid and fluid.
```
Comment (chleh) [inline ProcessLib/SmallDeformation/SmallDeformationProcessData.h]: There is no fluid here.

File: ProcessLib/SmallDeformation/SmallDeformationProcessData.h [hunk 4]
```diff
@@ -47,6 +56,12 @@ struct SmallDeformationProcessData
 
     std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
         material;
+    /// Solid's density. A scalar quantity, Parameter<double>.
```
Comment (chleh) [inline ProcessLib/SmallDeformation/SmallDeformationProcessData.h]: No need to repeat the type, I guess.

Comment (endJunction) [reply ProcessLib/SmallDeformation/SmallDeformationProcessData.h]: But it makes a nice link in the docu ;)
https://doxygen.opengeosys.org/ogs_file_param__prj__processes__process__HYDRO_MECHANICS__solid_density.html

Comment (chleh) [reply ProcessLib/SmallDeformation/SmallDeformationProcessData.h]: Touché, a bit at least.

**Comment by endJunction** [comment]
A test with two different densities is added and compared to analytical solution given by @nagelt .
dz
And the error is:
error

(The gaps in the graphs are rendering issues.)

**Comment by endJunction** [comment]
All jenkins jobs compiled just fine but the web job. @bilke working on the web build.

**Pull request merged**